sympy/functions/elementary/piecewise.py

```diff
@@ -195,242 +310,422 @@ def _eval_derivative(self, x):
+        # ------------------------------------------------------------------
+        try:
+            abei = self._abei(x)
+        except NotImplementedError:
+            from sympy import Integral
+            return Integral(self, (x, lo, hi))  # unevaluated
+
+        pieces = [(a, b) for a, b, _, _ in abei]
+        done = [(lo, hi, -1)]
+        oo = S.Infinity
+        for k, p in enumerate(pieces):
+            if p[:2] == (-oo, oo):
+                # all undone intervals will get this key
+                for j, (a, b, i) in enumerate(done):
+                    if i == -1:
+                        done[j] = a, b, k
+                break  # nothing else to consider
+            N = len(done) - 1
+            for j, (a, b, i) in enumerate(reversed(done)):
+                if i == -1:
+                    j = N - j
+                    done[j: j + 1] = _clip(p, (a, b), k)
+        # check for holes
+        for a, b, i in done:
+            if i == -1:
+                if a != b:
+                    # this requires a default (i == -1)
+                    # over a potentially non-zero interval
+                    # since a != b so we can't do this
+                    return S.NaN
+                else:
+                    # XXX are there expressions that would give
+                    # a value over a zero-width interval?
+                    # Integrating DiracDelta or Heaviside
+                    # does but we are not integrating right
+                    # now. If there are things to check for
+                    # check expression given by abei[i][-2]
+                    pass
+        # return the sum of the intervals
+        sum = S.Zero
+        for a, b, i in done:
+            sum += abei[i][-2]._eval_interval(x, a, b)
+        return _mmsimp(sum)
+
+    def _abei(self, sym):
+        """Return (a, b, e, i) where a and b are the lower and
+        upper bounds in which the expression e of argument i
+        in self is defined.
+
+        If there are any relationals not involving sym, or
+        any relational cannot be solved for sym, we raise
+        NotImplementedError. The evaluated conditions will
+        be returned as ranges. Discontinuous ranges will be
+        returned separately with identical expressions and
+        the first condition that evaluates to True will be
+        returned as the last tuple with a, b = -oo, oo.
+        """
         from sympy.solvers.inequalities import _solve_inequality
-        default = None
-        int_expr = []
+        from sympy.logic.boolalg import to_cnf, distribute_or_over_and
+
+        assert isinstance(self, Piecewise)
+
+        def _solve_relational(r):
+            rv = _solve_inequality(r, sym)
+            if isinstance(rv, Relational) and \
+                    sym in rv.free_symbols:
+                if rv.args[0] != sym:
+                    raise NotImplementedError(filldedent('''
+Unable to solve relational %s for %s.''' % (r, sym)))
```

Comment:
Indent this along with the rest of the code (isn't that the whole point of using filldedent?)